README.md

```diff
@@ -2,104 +2,116 @@
 
 [![test](https://github.com/mamba-org/provision-with-micromamba/workflows/test/badge.svg)](https://github.com/mamba-org/provision-with-micromamba/actions?query=workflow%3Atest)
 
-GitHub Action to provision a CI instance using micromamba
-
-## Inputs
-
-### `environment-file`
-
-**Optional** The the `environment.yml` file for the conda environment. Default is `environment.yml`
-
-### `environment-name`
-
-**Optional** Specify a custom environment name,
-to overwrite the name specified in the `environment.yml`,
-or in in case it was not specified in the `environment.yml`.
-
-### `micromamba-version`
-
-**Optional** Specifiy a custom micromamba version. Use `"latest"` for bleeding edge.
-
-### `extra-specs`
-
-**Optional** Specifiy additional specifications (packages) to install. Pretty useful when using matrix builds to pin versions of a test/run dependency.
-
-Note: for multiple packages, use multiline syntax (see examples below)
+GitHub Action to provision a CI instance using micromamba.
 
 ## Example usage
 
-Note: some shells need special syntax for invocation (e.g. `bash -l {0}`). You can set this up in [defaults](setup_default).
-
-```
-name: test
-on:
-  push: null
-
+```yaml
 jobs:
   test:
     runs-on: ubuntu-latest
-    name: test
     steps:
       - uses: actions/checkout@v2
 
-      - name: install mamba
+      - name: Install Conda environment from environment.yml
         uses: mamba-org/provision-with-micromamba@main
 
-      # linux and osx
-      - name: run python
+      # Linux and macOS
+      - name: Run Python
         shell: bash -l {0}
         run: |
           python -c "import numpy"
 
-      # windows
-      - name: run python
+      # Windows
+      # With Powershell:
+      - name: Run Python
         shell: powershell
         run: |
           python -c "import numpy"
-      - name: run cmd.exe
+      # Or with cmd:
+      - name: Run cmd.exe
         shell: cmd /C CALL {0}
         run: >-
           micromamba info && micromamba list
 ```
 
-## Example with customization
+> **Please** see the **[IMPORTANT](#IMPORTANT)** notes on additional information
+> on environment activation.
 
-```
-name: test
-on:
-  push: null
+## Example with customization
 
+```yaml
 jobs:
   test:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         pytest: ["6.1", "6.2"]
-    name: test
     steps:
       - uses: actions/checkout@v2
 
-      - name: install mamba
+      - name: Install Conda environment with Micromamba
         uses: mamba-org/provision-with-micromamba@main
         with:
           environment-file: myenv.yaml
-          environment-name: myenv
+          environment-name: myenvname
           extra-specs: |
             python=3.7
             pytest=${{ matrix.pytest }}
 ```
 
+## Example with download caching
+
+Use `cache-downloads` to enable download caching across action runs (`.tar.bz2` files).
```

Comment:
can we add up- and downsides of either way of caching? Which works better for which use case?

Reply:
Actually I'm not quite sure about those, given that we haven't used it a lot so far. My thoughts:

- Env caching should always work OK given how Conda works, and assuming that tar + untar doesn't lose file system permissions etc. One caveat is that you might have older versions of dependencies but that is limited to the cache TTL (currently hardcoded to 1 day). If you make any changes to your Conda env folder for whatever horrible reason, the changes are included in the cached version and your build may break.
- Download caching is generally slower because it only eliminates the download time. Actually it doesn't eliminate it, only (presumably) reduce it because it's loaded from a "faster" network (Azure Blob storage?!) Potentially it's even slower because download + extraction aren't done in parallel. In practice it always seems to be faster though. Advantage of download caching is that you are always up to date.

Other ideas?

Reply:
@jonashaag is the micromamba action also compatible with using Github's cache v3 action to cache mamba env?

Reply:
In my experience it seems that `cache-environment: true` leads to the environment being cache for a lot longer than a day: at least a week, probably longer.
This leads to updates of dependencies not being installed/updated in nightly test runs, which is a problem, so I had to disable it: https://github.com/geofileops/geofileops/pull/381

Reply:
Please note that this action is deprecated

Reply:
> Please note that this action is deprecated

I'm using `setup-micromamba@v1`, but I was sent here via the link in its README.md, below [When to use environment caching](https://github.com/mamba-org/setup-micromamba#notes-on-caching)

Reply:
Ah... I found the following issue that reports the same issue I'm facing: https://github.com/mamba-org/setup-micromamba/issues/50